pandas/_libs/index_class_helper.pxi.in

```diff
@@ -57,7 +57,10 @@ cdef class {{name}}Engine(IndexEngine):
             with warnings.catch_warnings():
                 # e.g. if values is float64 and `val` is a str, suppress warning
                 warnings.filterwarnings("ignore", category=FutureWarning)
-                indexer = values == val
+                if cmath.isnan(val):
```

Comment:
probably should use tempita to only do this check for FloatEngine

might be able to use something in the cnp namespace for the isnan check

Reply:
If we do this only for float, ``test_slice_locs_na`` raises from /indexes/test_base.py

Actually we don't need imports I think. Changed it

Reply:
`util.is_nan` might work

Reply:
Ah thx.